File: java/src/main/java/com/tormentnexus/claudecode/SystemPromptBuilder.java
package com.tormentnexus.claudecode;

// Re-implementation of Claude Code's Dynamic System Prompt Templating

import java.util.ArrayList;
import java.util.List;

public class SystemPromptBuilder {

    private String basePrompt = "You are TormentNexus, a highly capable AI assistant.";
    private final List<String> plugins = new ArrayList<>();

    public void setBasePrompt(String prompt) {
        this.basePrompt = prompt;
    }

    public void injectPluginPrompt(String prompt) {
        this.plugins.add(prompt);
    }

    public String build() {
        StringBuilder builder = new StringBuilder(this.basePrompt);
        if (!plugins.isEmpty()) {
            builder.append("\n\nAdditional Context / Plugin Instructions:\n");
            for (String plugin : plugins) {
                builder.append("- ").append(plugin).append("\n");
            }
        }
        return builder.toString().trim();
    }
}

File: csharp/ClaudeCodeFeatures/SystemPromptBuilder.cs
// Re-implementation of Claude Code's Dynamic System Prompt Templating

using System.Collections.Generic;
using System.Text;

namespace TormentNexus.ClaudeCodeFeatures
{
    public class SystemPromptBuilder
    {
        private string _basePrompt = "You are TormentNexus, a highly capable AI assistant.";
        private readonly List<string> _plugins = new List<string>();

        public void SetBasePrompt(string prompt)
        {
            _basePrompt = prompt;
        }

        public void InjectPluginPrompt(string prompt)
        {
            _plugins.Add(prompt);
        }

        public string Build()
        {
            var builder = new StringBuilder(_basePrompt);
            if (_plugins.Count > 0)
            {
                builder.AppendLine("\n\nAdditional Context / Plugin Instructions:");
                foreach (var plugin in _plugins)
                {
                    builder.AppendLine($"- {plugin}");
                }
            }
            return builder.ToString().TrimEnd();
        }
    }
}
